### PR TITLE
Fix resource manager clean up thread calling

### PR DESCRIPTION
**Pull request by YuanTingHsieh**
- Fix event firing
- Fix a situation where `task_id` might be referenced before assignment in `_process_task_request` in `nvflare/private/fed/server/fed_server.py`
- Fix client replies error handling in `nvflare/private/fed/server/job_runner.py`
- Fix return responses in `ClientEngine`

### CODE REVIEW

File: nvflare/private/fed/client/client_engine.py [hunk 1]
```diff
@@ -160,7 +160,7 @@ def start_app(
             resource_manager,
         )
 
-        return "Start the client app..."
+        return ""
```
Comment (yhwen) [inline nvflare/private/fed/client/client_engine.py:163]: This message is sending back to the admin console to display. Need to keep it.

File: nvflare/private/fed/client/client_engine.py [hunk 2]
```diff
@@ -201,7 +201,7 @@ def abort_app(self, run_number: str) -> str:
 
         self.client_executor.abort_train(self.client, run_number)
 
-        return "Abort signal has been sent to the client App."
+        return ""
```
Comment (yhwen) [inline nvflare/private/fed/client/client_engine.py:204]: same need to keep the message to show to admin.

File: nvflare/private/fed/client/client_engine.py [hunk 3]
```diff
@@ -213,7 +213,7 @@ def abort_task(self, run_number: str) -> str:
 
         self.client_executor.abort_task(self.client, run_number)
 
-        return "Abort signal has been sent to the current task. "
+        return ""
```
Comment (yhwen) [inline nvflare/private/fed/client/client_engine.py:216]: keep the message,

File: nvflare/private/fed/client/client_engine.py [hunk 4]
```diff
@@ -222,7 +222,7 @@ def shutdown(self) -> str:
         self.fire_event(EventType.SYSTEM_END, self.new_context())
         _ = self.executor.submit(lambda p: _shutdown_client(*p), [self.client, self.admin_agent, touch_file])
 
-        return "Shutdown the client..."
+        return ""
```
Comment (yhwen) [inline nvflare/private/fed/client/client_engine.py:225]: keep the message.

File: nvflare/private/fed/client/client_engine.py [hunk 5]
```diff
@@ -231,7 +231,7 @@ def restart(self) -> str:
         self.fire_event(EventType.SYSTEM_END, self.new_context())
         _ = self.executor.submit(lambda p: _shutdown_client(*p), [self.client, self.admin_agent, touch_file])
 
-        return "Restart the client..."
+        return ""
```
Comment (yhwen) [inline nvflare/private/fed/client/client_engine.py:234]: keep the message.

File: nvflare/private/fed/client/client_engine.py [hunk 6]
```diff
@@ -245,7 +245,7 @@ def delete_run(self, run_num: int) -> str:
         run_number_folder = os.path.join(self.args.workspace, WorkspaceConstants.WORKSPACE_PREFIX + str(run_num))
         if os.path.exists(run_number_folder):
             shutil.rmtree(run_number_folder)
-        return "Delete run folder: {}".format(run_number_folder)
+        return ""
```
Comment (yhwen) [inline nvflare/private/fed/client/client_engine.py:248]: keep the message.

**Review by yhwen — COMMENTED** [review]
The check_client_replies() change looks okay. But this requires to tests all sorts scenarios, including the app failed to deploy, failed to start, etc. And check the proper messages have been logged.

File: nvflare/private/fed/server/job_runner.py [hunk 7]
```diff
@@ -277,8 +285,8 @@ def restore_running_job(self, run_number: str, job_id: str, job_clients, snapsho
             job = job_manager.get_job(jid=job_id, fl_ctx=fl_ctx)
             with self.lock:
                 self.running_jobs[run_number] = job
-        except:
-            self.log_error(fl_ctx, f"Failed to restore the job:{job_id} to the running job table.")
+        except Exception as e:
+            self.log_error(fl_ctx, f"Failed to restore the job: {job_id} to the running job table: {e}.")
```
Comment (chesterxgchen) [inline nvflare/private/fed/server/job_runner.py:289]: do you need the stack trace ?  exec_info = True ?

File: nvflare/private/fed/utils/fed_utils.py [hunk 8]
```diff
@@ -93,3 +95,18 @@ def listen_command(listen_port, engine, execute_func, logger):
...
         if listener:
             listener.close()
+
+
+def check_client_replies(replies, client_sites: List[str], command: str):
+    display_sites = ",".join(client_sites)
+    if not replies:
+        raise RuntimeError(f"Failed to {command} to the clients {display_sites}: no replies.")
+    if len(replies) != len(client_sites):
+        raise RuntimeError(f"Failed to {command} to the clients {display_sites}: not enough replies.")
```
Comment (chesterxgchen) [inline nvflare/private/fed/utils/fed_utils.py:105]: does replies has site information ? if yes, it might be better to tell which site did not get reply

**Review by chesterxgchen — APPROVED** [review]
overall looks good to me. maybe need more tests
add few minor comments

**Review by yhwen — APPROVED** [review]

File: nvflare/private/fed/server/job_runner.py (hunk 7 above)
Comment (YuanTingHsieh) [reply nvflare/private/fed/server/job_runner.py:289]: this is not logger, we could add that into general log_error though

**Pull request merged**